src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimMatcher.java

```diff
@@ -66,4 +90,149 @@ public boolean hasResponseDeadlineBeenExtended() {
+    public boolean isMoreDetailsRequired() {
+        Optional<LocalDateTime> lastOrder = getTimeOfLastNonSDOOrder();
+        Optional<LocalDateTime> sdoTime = getSDOTime();
+        return sdoTime.isPresent()
+            && caseData.getHearingDate() == null
+            && featureToggleService.isCaseProgressionEnabled()
+            && (lastOrder.isEmpty()
+            || lastOrder.get().isBefore(sdoTime.get()));
+    }
+
+    @Override
+    public boolean isHearingLessThanDaysAway(int i) {
+        return caseData.getHearingDate() != null
+            && LocalDate.now().plusDays(i + 1L).isAfter(caseData.getHearingDate());
+    }
+
+    @Override
+    public Optional<LocalDate> getHearingDate() {
+        return Optional.ofNullable(caseData.getHearingDate());
+    }
+
+    @Override
+    public Optional<LocalDateTime> getBundleCreationDate() {
+        if (caseData.getHearingDate() != null) {
+            return caseData.getCaseBundles().stream()
+                .filter(b -> b.getValue().getCreatedOn().isPresent()
+                    && b.getValue().getBundleHearingDate()
+                    .map(d -> caseData.getHearingDate().equals(d)).orElse(Boolean.FALSE))
+                .map(d -> d.getValue().getCreatedOn().orElse(null))
+                .max(LocalDateTime::compareTo);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<LocalDateTime> getWhenWasHearingScheduled() {
+        return getMostRecentEventOfType(EnumSet.of(CaseEvent.HEARING_SCHEDULED))
+            .map(CaseEventDetail::getCreatedDate);
+    }
+
+    protected Optional<LocalDateTime> getTimeOfMostRecentEventOfType(Set<CaseEvent> events) {
+        return getMostRecentEventOfType(events)
+            .map(CaseEventDetail::getCreatedDate);
+    }
+
+    protected Optional<CaseEventDetail> getMostRecentEventOfType(Set<CaseEvent> events) {
+        Set<String> eventNames = events.stream().map(CaseEvent::name).collect(Collectors.toSet());
+        return eventHistory.stream()
+            .filter(e -> eventNames.contains(e.getId()))
+            .findFirst();
+    }
+
+    @Override
+    public boolean isTrialArrangementStatusActive() {
+        int dayLimit = 6 * 7;
+        Optional<LocalDate> hearingDate = getHearingDate();
+        if (caseData.isFastTrackClaim()
+            && hearingDate.isPresent()
+            && isHearingLessThanDaysAway(6 * 7)) {
+            Optional<LocalDateTime> lastOrder = getTimeOfLastNonSDOOrder();
+            return lastOrder.isEmpty()
+                || hearingDate.get().minusDays(dayLimit)
+                .isAfter(lastOrder.get().toLocalDate());
+        } else {
+            return false;
+        }
+    }
+
+    @Override
+    public boolean isTrialScheduledNoPaymentStatusActive() {
+        Optional<LocalDateTime> hearingScheduledDate = getWhenWasHearingScheduled();
+        Optional<LocalDateTime> orderDate = getTimeOfLastNonSDOOrder();
+        return !isHearingLessThanDaysAway(6 * 7)
```

Comment:
Given how much 6 * 7 is used, I feel it should be a constant.

Reply:
Should be resolved now